### ISSUE TITLE
Cloud Custodian on Windows - PermissionError: [Errno 13] Permission denied: 'C:\\Users\\xxxxx\\AppData\\Local\\Temp\\tmp40l2leln

### ISSUE THREAD

**Issue by nirisha** [cloud-custodian/cloud-custodian#3681]
Getting the below error when trying to run a policy on S3 policy on  a Windows 10 machine. Cloud Custodian has been working fine so far for EC2 policies, so it is not like Cloud Custodian itself is not working on my Windows 10 machine. Tried the same policy on a MacOS and it ran fine.. 

_Windows user-id information removed from below purposefully.._

**Error:**
```
c:\Users\xxxx\Desktop\xxxx\Custodian>custodian run -s . set-s3-new-bucket-standards.yml
2019-03-14 14:24:11,641: custodian.policy:INFO Provisioning policy lambda s3-configure-standards-real-time
2019-03-14 14:24:12,144: custodian.output:ERROR Error while executing policy
Traceback (most recent call last):
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\policy.py", line 487, in provision
    role=self.policy.options.assume_role)
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 283, in publish
    func, role, s3_uri, qualifier=alias)
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 405, in _create_or_update
    code_ref = {'ZipFile': archive.get_bytes()}
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 216, in get_bytes
    return open(self._temp_archive_file.name, 'rb').read()
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\xxxx\\AppData\\Local\\Temp\\tmp40l2leln'
2019-03-14 14:24:12,152: custodian.commands:ERROR Error while executing policy s3-configure-standards-real-time, continuing
Traceback (most recent call last):
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\commands.py", line 237, in run
    policy()
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\policy.py", line 868, in __call__
    resources = mode.provision()
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\policy.py", line 487, in provision
    role=self.policy.options.assume_role)
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 283, in publish
    func, role, s3_uri, qualifier=alias)
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 405, in _create_or_update
    code_ref = {'ZipFile': archive.get_bytes()}
  File "c:\users\xxxx\appdata\local\programs\python\python36\lib\site-packages\c7n\mu.py", line 216, in get_bytes
    return open(self._temp_archive_file.name, 'rb').read()
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\xxxx\\AppData\\Local\\Temp\\tmp40l2leln'
```

**Policy:**
```
policies:

   - name: s3-configure-standards-real-time
     resource: s3
     description: |
       This policy is triggered when a new S3 bucket is created and it applies
       the AWS AES256 Default Bucket Encryption, Tags the creators ID, enables
       object versioning, configures the bucket lifecycle and enables logging.
     mode:
       type: cloudtrail
       events:
         - CreateBucket
       role: arn:aws:iam::{account_id}:role/Cloud_Custodian_S3_Lambda_Role
       timeout: 200
     actions:
       - type: auto-tag-user
         tag: CreatorName
       - type: set-bucket-encryption
       - type: toggle-versioning
         enabled: true
       - type: toggle-logging
         target_bucket: "{account_id}-{region}-s3-logs"
         target_prefix: "{source_bucket_name}/"
```

**Comment by kapilt** [comment]
Specifically custodian windows support around serverless policies has been historically problematic, there have been some attempts to rectify this but most of the contributors don't use windows. Recommend would be to use a linux server/vm, osx machine, or docker image.

@stefangordon @logachev does this error look familiar ^?

**Comment by stefangordon** [comment]
Yep familiar.  I think we attempted to fix that by changing the open command a bit at
https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/mu.py#L230

But looks like there might be another open in there that wasn't modified - so possibly an easy fix.

**Comment by kapilt** [comment]
i believe this is fixed with #3842 we run unit tests on windows as part of ci now, and a number of fixes were made to get there.
